lib/extract/netcdf/kernel_data_netcdf.f90

```diff
@@ -0,0 +1,278 @@
+        integer, intent(in) :: value
+        integer :: retval
+        retval = CheckError(nf90_put_var(this%ncid, this%var_id(this%next_var_index),  &
+                                         value))
+        this%next_var_index = this%next_var_index + 1
+    end subroutine WriteScalarInteger
+
+    ! -------------------------------------------------------------------------
+    subroutine ReadScalarInteger(this, name, value)
+        use netcdf
+        implicit none
+
+        class(PSyDataType), intent(inout) :: this
+        character(*), intent(in) :: name
+        integer :: value
+        integer :: retval, varid
+        retval = CheckError(nf90_inq_varid(this%ncid, name, varid))
+        retval = CheckError(nf90_get_var(this%ncid, varid, value))
+    end subroutine ReadScalarInteger
+
+    ! -------------------------------------------------------------------------
+    subroutine DeclareScalarReal(this, name, value)
+        use netcdf
+        implicit none
+        class(PSyDataType), intent(inout) :: this
+        character(*), intent(in) :: name
+        real, intent(in) :: value
+        integer :: retval
+        retval = CheckError(nf90_def_var(this%ncid, name, Nf90_REAL,     &
+                                         this%var_id(this%next_var_index)))
+        this%next_var_index = this%next_var_index + 1
+        call CheckLastDeclaration(this)
+    end subroutine DeclareScalarReal
+
+    ! -------------------------------------------------------------------------
+    subroutine WriteScalarReal(this, name, value)
+        use netcdf
+        implicit none
+        class(PSyDataType), intent(inout) :: this
+        character(*), intent(in) :: name
+        real, intent(in) :: value
+        integer :: retval
+        retval = CheckError(nf90_put_var(this%ncid, this%var_id(this%next_var_index),  &
+                                         value))
+        this%next_var_index = this%next_var_index + 1
+    end subroutine WriteScalarReal
+
+    ! -------------------------------------------------------------------------
+    subroutine ReadScalarReal(this, name, value)
+        use netcdf
+        implicit none
+
+        class(PSyDataType), intent(inout) :: this
+        character(*), intent(in) :: name
+        real :: value
+        integer :: retval, varid
+        retval = CheckError(nf90_inq_varid(this%ncid, name, varid))
+        retval = CheckError(nf90_get_var(this%ncid, varid, value))
+    end subroutine ReadScalarReal
+
+    ! -------------------------------------------------------------------------
+    subroutine WriteScalarDoublePrecision(name, value)
+        implicit none
+        character(*), intent(in) :: name
+        double precision, intent(in) :: value
+    end subroutine WriteScalarDoublePrecision
+    
+    ! -------------------------------------------------------------------------
+    subroutine DeclareFieldDouble(this, name, value, prefix)
+        use netcdf
+        use field_mod, only : r2d_field
```

Comment:
This is dl_esm_inf specific but there's nothing in the naming scheme of this utility library to indicate that this is an issue. Maybe it should be in `lib/extract/gocean/netcdf` or `netcdf/gocean`?

Reply:
Ah, I missed that post. You suggest gocean instead of dl_esm_inf - more options. I would somewhat favour dl_esm inf: it could (theoretically) be possible to write a different infrastructure library that would work with the gocean API, but would not be compatible with dl_esm_inf.

Reply:
Done, but I used ``lib/extract/dl_esm_inf/netcdf`` - I think you were ok with that?